insn_test.go

```diff
@@ -596,28 +603,39 @@ func TestTAX(t *testing.T) {
 	checkBackends(t, filter, []byte{}, XDPDrop)
 }
 
+// XDPDrop if packet matches filter, XDPPass otherwise
+type backend func(testing.TB, []bpf.Instruction, []byte) XDPAction
+
 // checkBackends builds an eBPF program using each backend, and checks it returns the correct action
 // Input packet is 0 padded to min ethernet length, and output is checked to be unchanged
-func checkBackends(tb testing.TB, filter []bpf.Instruction, in []byte, res XDPAction) {
-	tb.Helper()
+func checkBackends(t *testing.T, filter []bpf.Instruction, in []byte, expected XDPAction) {
+	t.Helper()
+
+	// -short skips tests that require Linux
```

Comment:
You could use runtime.GOOS to skip in that case. https://golang.org/pkg/runtime/#pkg-constants

Reply:
Good point. I re-worded this to mention permissions only, as that's the most interesting case (and it can't be detected using `GOOS`).